Client/Experiments/Messaging/MessagingManager.swift

```diff
@@ -0,0 +1,257 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+import MozillaAppServices
+import Shared
+
+/// `Messaging` is synonymous with `GleanPlumb`.
+
+protocol MessagingManagable { }
+
+extension MessagingManagable {
+    var messagingManager: MessagingManager {
+        return MessagingManager.shared
+    }
+}
+
+protocol MessagingManagerProvider {
+    
+    /// Does the bookkeeping and preparation of messages for their respective surfaces.
+    /// Deprecate.
+    func onStartup()
+    
+    /// Finds the next message to be displayed out of all showable messages.
+    /// Surface calls.
+    func getNextMessage(for surface: MessageSurfaceId) -> Message?
+    
+    /// Report impressions in Glean, and then pass the bookkeeping to increment the impression count and expire to `MessageStore`.
+    /// Surface calls.
+    func onMessageDisplayed(message: Message)
+    
+    /// Using the helper, this should get the message action string ready for use.
+    /// Surface calls.
+    func onMessagePressed(message: Message)
+    
+    /// Handles what to do with a message when a user has dismissed it.
+    /// Surface calls.
+    func onMessageDismissed(message: Message)
+    
+    /// If the message is malformed (missing key elements the surface expects), then
+    /// report the malformed message.
+    /// Manager calls.
+    func onMalformedMessage(messageKey: String)
+}
+
+/// The `MessagingManager` is responsible for several things including:
+/// - preparing Messages for a given UI surface
+/// - reacting to user interactions on the `Message` to handle:
+///     - impression counts
+///     - user dismissal of a message
+///     - expiration logic
+/// - reporting telemetry for `Message`s
+class MessagingManager: MessagingManagerProvider, MessagingHelperProtocol, MessageStoreProvider, Loggable {
```

Comment:
Let's see if we can break some of these functions in smaller chunks to improve legibility.